corporate/lib/stripe.py

```diff
@@ -1183,6 +1188,104 @@ def make_end_of_cycle_updates_if_needed(
 
             if plan.status == CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE:
                 self.process_downgrade(plan)
+
+            if plan.status == CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_ANNUAL_TO_MONTHLY_CYCLE:
```

Comment:
So the flow with these changes is ...

1. `update_plan` calls `BillingSession.do_update_plan`
2. `do_update_plan` calls `make_end_of_cycle_updates_if_needed` with no changes to the existing plan
3. There's a successful call to `do_change_plan_status`, which does not create the new plan and sends a success message from the server to the client
4. The redirect on the client side with the success response text gets the billing page information again, which triggers the server to end the current plan and create a new plan with the correct billing cycle

It feels ... indirect that the server hasn't completed the update to a plan with a new billing cycle until a new call to the billing page happens.

Possible alternatives:
- Instead of using `make_end_of_cycle_updates` for the new plan creation and ending the old plan, maybe there's a `SWITCH_FREE_TRIAL_BILLING_CYCLE_NOW` status and a separate, immediate change via a new BillingSession function that's called in `do_update_plan` for that status change.
- Call `make_end_of_cycle_updates_if_needed` (again) after `do_change_plan_status` for these two cases in `do_update_plan`.